### PR TITLE
Open Governance Model

### LINKED ISSUE #4683: Add Open Governance File
As part of joining the [HPSF](https://hpsf.io/), we will document our governance model, mirroring our initial technical charter, in a `GOVERNANCE.rst` file. We will also cross-reference it with the existing `CONTRIBUTING.rst` file, especially for write/merge permissions.

### PR DESCRIPTION
**Pull request by ax3l**
This is a first draft of the open governance model for WarpX, developed after discussion with the current leadership team.

The goal of the governance model is to:
- document and make transparent our decision structure,
- structure and openly credit sensible roles,
- clarify appointment and off-boarding,
- encourage participation, engagement and growth.

Thoughts that guided the initial document include:
- accurately reflecting our successful project workflows,
- evolving it for shared responsibility and co-ownership,
- avoiding unnecessary overheads and deadlocks,
- be flexible enough to enable later introduction of sub-roles and further formalization, depending on community growth.

Once reviewed and accepted, this implements and closes #4683

## Next Steps

- [x] discussion and revision of the draft between @ECP-WarpX/warpx-admins, March 13th, 2024
- [x] presentation at the weekly developer meeting,  March 26th, 2024
- [x] discussion, revision and ratification at the weekly developer meeting, April 23rd, 2024

### CODE REVIEW

File: GOVERNANCE.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+- Jean-Luc Vay (chair)
+- Remi Lehe
+- Axel Huebl
+
+See: `GitHub team <https://github.com/orgs/ECP-WarpX/teams/warpx-admins>`__
+
+Role
+^^^^
+
+The steering committee (SC) can change repo requirements, can do administrative operations such as rename/move/archive projects, etc.
```
Comment (RemiLehe) [inline GOVERNANCE.rst]: The term `repo requirements` may need to be clarified/defined.

File: GOVERNANCE.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+SC members call votes for decisions.
```
Comment (RemiLehe) [inline GOVERNANCE.rst]: Do you mean "can call for a vote"?

File: GOVERNANCE.rst [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+The SC can veto decisions of the technical committee (TC) by voting in the SC.
+
+The SC can change the governance structure, but only in an unanimous vote.
+
+Decision Process
+^^^^^^^^^^^^^^^^
+
+Decision are made in a public, majority vote of SC members.
```
Comment (RemiLehe) [inline GOVERNANCE.rst]: Do we need to specify how members will be notified of a vote (e.g. one week in advance email notice), and the voting procedure (e.g. Zoom meeting, or email)?

File: GOVERNANCE.rst (hunk 2 above)
Comment (RemiLehe) [reply GOVERNANCE.rst]: Also, here, does "decisions" refer to governance-related decision, or technical decisions?

File: GOVERNANCE.rst (hunk 1 above)
Comment (ax3l) [reply GOVERNANCE.rst]: - GitHub organization administration
- repository settings
- branch protection rules
```suggestion
The steering committee (SC) can change organizational settings, do administrative operations such as rename/move/archive repositories, change branch protection rules, etc.
```

File: GOVERNANCE.rst (hunk 3 above)
Comment (ax3l) [reply GOVERNANCE.rst]: Good questions, we tried to keep procedure slick for now but for the SC a bit more structure due to its weight might be useful. Maybe the simplest thing is: everyone in the SC has like 1-2 weeks to vote if a vote is called.

```suggestion
Decision are made in a non-confidential, majority vote of SC members.
```

File: GOVERNANCE.rst (hunk 2 above)
Comment (ax3l) [reply GOVERNANCE.rst]:
```suggestion
SC members can call votes for decisions (technical of governance).
```

File: GOVERNANCE.rst [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+See: `GitHub team <https://github.com/orgs/ECP-WarpX/teams/warpx-push-merge-write>`__
+
+Role
+^^^^
+
+The technical committee (TC) is the core governance body, where under normal operations most ideas are discussed and decisions are made.
+Individual TC members can approve and merge code changes.
+Usually, they seek approval by another maintainer for their own changes, too.
+TC members lead and weight in on technical discussions and can call a vote between TC members, if needed for a technical decision.
```
Comment (ax3l) [inline GOVERNANCE.rst]:
```suggestion
TC members lead - and weigh in on - technical discussions and, if needed, can call for a vote between TC members for a technical decision.
```

File: GOVERNANCE.rst [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+TC members merge/close PRs and issues, and moderate (including block/mute) bad actors.
+The TC can propose governance changes to the SC.
+
+
+Decision Process
+^^^^^^^^^^^^^^^^
+
+Discussion on a topic in the weekly developer meetings.
```
Comment (ax3l) [inline GOVERNANCE.rst]:
```suggestion
Discussion in the TC usually happens in the weekly developer meetings.
```

File: GOVERNANCE.rst [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+If someone calls for a vote to make a decision: majority based on the passed votes; we need 50% of the committee participating to vote. In the absence of a quorum, the SC will decide.
+
+Decision are made in a public, majority vote of TC members.
```
Comment (ax3l) [inline GOVERNANCE.rst]:
```suggestion
Votes are cast in non-confidential manner.
```

File: GOVERNANCE.rst [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+TC members merge/close PRs and issues, and moderate (including block/mute) bad actors.
+The TC can propose governance changes to the SC.
+
+
+Decision Process
+^^^^^^^^^^^^^^^^
+
+Discussion on a topic in the weekly developer meetings.
+
+If someone calls for a vote to make a decision: majority based on the passed votes; we need 50% of the committee participating to vote. In the absence of a quorum, the SC will decide.
```
Comment (ax3l) [inline GOVERNANCE.rst]:
```suggestion
If someone calls for a vote to make a decision: majority based on the cast votes; we need 50% of the committee participating to vote. In the absence of a quorum, the SC will decide according to its voting rules.
```

File: GOVERNANCE.rst [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+Decision are made in a public, majority vote of TC members.
+
+In the absence of a quorum (>50%) in the TC, the SC decides according to its voting rules.
+
```
Comment (ax3l) [inline GOVERNANCE.rst]: Duplicate.
```suggestion
```

File: GOVERNANCE.rst [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+TC members can individually highlight  new contributors, unless a vote is called on an individual.
```
Comment (ax3l) [inline GOVERNANCE.rst]:
```suggestion
TC members can individually appoint new contributors, unless a vote is called on an individual.
```

File: GOVERNANCE.rst [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+
+Appointment Process
+^^^^^^^^^^^^^^^^^^^
+
+TC members are the maintainers of WarpX.
+As a TC member, regularly attending and contributing to the weekly developer meetings is expected.
+
+One is appointed to the TC by the steering committee, in an unanimous vote.
```
Comment (ax3l) [inline GOVERNANCE.rst]:
```suggestion
One is appointed to the TC by the steering committee, in a unanimous vote, or by majority vote of the TC. The SC can veto appointments.
```

File: GOVERNANCE.rst [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+Current Roster
+^^^^^^^^^^^^^^
+
+See: `GitHub team <https://github.com/orgs/ECP-WarpX/teams/warpx-contributors>`__
+
+Role
+^^^^
+
+Contributors are valuable, vetted developers to WarpX.
```
Comment (ax3l) [inline GOVERNANCE.rst]:
```suggestion
Contributors are valuable, vetted developers of WarpX.
```

File: GOVERNANCE.rst [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+Contributions can be in many forms and not all need to be code contributions.
+Examples include code pull requests, support in issues & user discussions, writing and updating documentation, R&D on novel numerics, testing and benchmarking, etc.
+Contributors can participate in developer meetings and weight in on discussions.
```
Comment (ax3l) [inline GOVERNANCE.rst]:
```suggestion
Contributors can participate in developer meetings and weigh in on discussions.
```

File: GOVERNANCE.rst [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+^^^^^^^^^^^^^^
+
+See: `GitHub team <https://github.com/orgs/ECP-WarpX/teams/warpx-contributors>`__
+
+Role
+^^^^
+
+Contributors are valuable, vetted developers to WarpX.
+Contributions can be in many forms and not all need to be code contributions.
+Examples include code pull requests, support in issues & user discussions, writing and updating documentation, R&D on novel numerics, testing and benchmarking, etc.
```
Comment (ax3l) [inline GOVERNANCE.rst]:
```suggestion
Examples include code pull requests, support in issues & user discussions, writing and updating documentation, writing tutorials, visualizations, R&D on algorithms, testing and benchmarking, etc.
```

File: GOVERNANCE.rst [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+^^^^^^^^^^^^^^^^
+
+Contributors can individually decide on classification (triage) of pull requests, issues, and GitHub discussion pages.
+
+Appointment Process
+^^^^^^^^^^^^^^^^^^^
+
+Appointed after contributing to WarpX (see above) by any member of the TC.
+
+The role can be lost by resigning or majority vote by TC or SC due to inactivity, bad acting or other.
```
Comment (ax3l) [inline GOVERNANCE.rst]:
```suggestion
The role can be lost by resigning or by decision of an individual TC or SC member, e.g., due to inactivity, bad acting or other.
```

File: GOVERNANCE.rst [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+- Jean-Luc Vay (chair)
+- Remi Lehe
+- Axel Huebl
+
+See: `GitHub team <https://github.com/orgs/ECP-WarpX/teams/warpx-admins>`__
+
+Role
+^^^^
+
+The steering committee (SC) can change organizational settings, do administrative operations such as rename/move/archive repositories, change branch protection rules, etc.
```
Comment (ax3l) [inline GOVERNANCE.rst]: Clarified per discussion:
```suggestion
Members of the steering committee (SC) can change organizational settings, do administrative operations such as rename/move/archive repositories, change branch protection rules, etc.
```
SC members can change branch protection rules, etc. as needed for daily administrative work (e.g., CI).

File: GOVERNANCE.rst [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+
+
+Technical Committee
+-------------------
+
+Current Roster
+^^^^^^^^^^^^^^
+
+- Luca Fedeli
+- Roelof Groenwald
```
Comment (roelof-groenewald) [inline GOVERNANCE.rst]: 😄 
```suggestion
- Roelof Groenewald
```

Comment (ax3l) [reply GOVERNANCE.rst]: Ooops! Sorry :)

Comment (roelof-groenewald) [reply GOVERNANCE.rst]: No worries! Thanks for updating.

File: GOVERNANCE.rst [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+
+See: `GitHub team <https://github.com/orgs/ECP-WarpX/teams/warpx-admins>`__
+
+Role
+^^^^
+
+Members of the steering committee (SC) can change organizational settings, do administrative operations such as rename/move/archive repositories, change branch protection rules, etc.
+SC members can call votes for decisions (technical or governance).
+
+The SC can veto decisions of the technical committee (TC) by voting in the SC.
```
Comment (ax3l) [inline GOVERNANCE.rst:27]: Suggestion from feedback to add more trust to the TC:
```suggestion
The SC can veto decisions of the technical committee (TC) by voting in the SC.
The TC can overwrite a veto with a 2/3rd majority vote in the TC.
```

Comment (ax3l) [reply GOVERNANCE.rst:27]: Approved today by @WeiqunZhang @atmyers @RemiLehe in meeting.

Comment (ax3l) [reply GOVERNANCE.rst:27]: Ok'd also by @jlvay via Slack.

File: GOVERNANCE.rst [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+The TC can overwrite a veto with a 2/3rd majority vote in the TC.
```
Comment (ax3l) [inline GOVERNANCE.rst:28]: Here and in TC add:
```suggestion
The TC can overwrite a veto with a 2/3rd majority vote in the TC.
Decisions are documented in the `weekly developer meeting notes <https://docs.google.com/document/d/1eYD8EYCYDI0H7FhiiEuRUDJZ5pPDr6MUL-IibE-Pk50/edit>`__ and/or on the GitHub repository.
```

File: GOVERNANCE.rst [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+
+
+Decision Process
+^^^^^^^^^^^^^^^^
+
+Discussion in the TC usually happens in the weekly developer meetings.
+
+If someone calls for a vote to make a decision: majority based on the cast votes; we need 50% of the committee participating to vote. In the absence of a quorum, the SC will decide according to its voting rules.
+
+Votes are cast in a non-confidential manner.
```
Comment (ax3l) [inline GOVERNANCE.rst:89]:
```suggestion
Votes are cast in a non-confidential manner.
Decisions are documented in the `weekly developer meeting notes <https://docs.google.com/document/d/1eYD8EYCYDI0H7FhiiEuRUDJZ5pPDr6MUL-IibE-Pk50/edit>`__ and/or on the GitHub repository.
```

**Review by n01r — APPROVED** [review]

**Review by dpgrote — APPROVED** [review]
This looks good to me.
Though, a question is whether additional approvals are needed to change this document beyond the normal TC code change approval process.

File: GOVERNANCE.rst [hunk 20]
```diff
@@ -0,0 +1,143 @@
...
+
+- Luca Fedeli
+- Roelof Groenewald
+- David Grote
+- Axel Huebl
+- Revathi Jambunathan
+- Remi Lehe
+- Andrew Myers
+- Maxence Thévenet
+- Weiqun Zhang
```
Comment (ax3l) [inline GOVERNANCE.rst:66]: Added after review from @jlvay and @AlexanderSinn :
```suggestion
- Jean-Luc Vay
- Weiqun Zhang
```

**Comment by ax3l** [comment]
Thanks @dpgrote.

> a question is whether additional approvals are needed to change this document beyond the normal TC code change approval process.

Changes to this document would be approved by SC.
> The SC can change the governance structure, but only in a unanimous vote.

**Review by jlvay — APPROVED** [review]
Looks good.

**Review by ax3l — COMMENTED** [review]
I approve.

**Review by dpgrote — APPROVED** [review]
Looks good

**Review by RemiLehe — APPROVED** [review]

**Review by roelof-groenewald — APPROVED** [review]

**Review by atmyers — APPROVED** [review]

**Review by lucafedeli88 — APPROVED** [review]

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**